app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java

```diff
@@ -137,7 +138,7 @@ private void handlePaymentSuccess(PaymentData paymentData) {
 
         // getPaymentMethodToken will only return null if PaymentMethodTokenizationParameters was
         // not set in the PaymentRequest.
-        if (token != null) {
+        if ((token != null) && (paymentData.getCardInfo().getBillingAddress() != null)) {
```

Comment:
Similar to the above - if the billing address is required in the request, but is null, this is a serious problem and the error shouldn't be swallowed. On the other hand, if the billing address requirement is no longer there, we want this part to be removed from the handler.

Most importantly, since this is a sample, we don't want to give the wrong idea that the token is somehow invalid if the billing address is not set. If you really feel this check is required, perhaps you can just have it display "unknown" if the billing address is not present. Something like this:

```
if (token != null) {
    String billingName = "Unknown";
    if (paymentData.getCardInfo().getBillingAddress() != null) {
        billingName = paymentData.getCardInfo().getBillingAddress().getName();
    }
    Toast.makeText(this, getString(R.string.payments_show_name, billingName), Toast.LENGTH_LONG).show();
}
```